pkg/reconciler/common/reconcile.go

```diff
@@ -82,9 +109,15 @@ func (r *GenericDeploymentReconciler) resolveSinkURL(ctx context.Context) (*apis
 }
 
 // reconcileAdapter reconciles the state of the source's adapter.
-func (r *GenericDeploymentReconciler) reconcileAdapter(ctx context.Context, desiredAdapter *appsv1.Deployment) error {
+func (r *GenericDeploymentReconciler) reconcileAdapter(ctx context.Context,
+	desiredAdapter *appsv1.Deployment, rbacOwners []kmeta.OwnerRefable) error {
+
 	src := v1alpha1.SourceFromContext(ctx)
 
+	if err := r.reconcileRBAC(ctx, rbacOwners); err != nil {
+		return fmt.Errorf("reconciling RBAC objects: %w", err)
+	}
+
```

Comment:
With that interface implemented, we can now plug some new `reconcileRBAC()` logic inside the top-level `reconcileAdapter()` function. I did it that way because I consider RBAC objects as _part of_ the adapter (they have no use without adapter, and the adapter doesn't work without them).